axum/src/routing/route.rs

```diff
@@ -171,16 +168,6 @@ where
             }
         };
 
-        if res
-            .extensions()
-            .get::<AlreadyPassedThroughRouteFuture>()
-            .is_some()
-        {
-            return Poll::Ready(Ok(res));
-        } else {
-            res.extensions_mut().insert(AlreadyPassedThroughRouteFuture);
-        }
```

Comment:
tbh I'm not totally sure why everything still worked after removing this. Doesn't seem related to this change because it also works on `main` if you remove this. So must be something else that changed 🤷 

https://github.com/tokio-rs/axum/pull/755 did add a regression test and I also tried the repro from https://github.com/tokio-rs/axum/issues/747. Both worked correctly.